src/microbe_stage/editor/PatchMapNode.cs

```diff
@@ -96,11 +123,23 @@ public override void _GuiInput(InputEvent @event)
                 Pressed: true, ButtonIndex: (int)ButtonList.Left or (int)ButtonList.Right,
             })
         {
+            ((PatchMapDrawer)GetParent()).MarkDirty();
```

Comment:
Explicitly referring to the parent node is an antipattern, especially casting to a special type. There should instead be emitted a signal in this case, which the parent node sets up.